client/src/components/Modal/tests/Modal-test.js

```diff
@@ -0,0 +1,89 @@
+/* global jest, test, describe, it, expect */
+
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import Modal from '../Modal';
+
+let props;
+
+/** Component that printouts all of its props */
+const PrintProps = (componentName) => ({ children, ...propsToPrint }) => (
+  <div>
+    <div>{componentName}</div>
+    <ul>
+      {Object
+        .entries(propsToPrint)
+        .map(([key, value]) => (<li key={key}>{key}: {value.toString()}</li>))
+      }
+    </ul>
+    {children}
+  </div>
+);
+
+beforeEach(() => {
+  props = {
+    isOpen: true,
+    className: 'my-modal',
+    modalClassName: 'my-modal-dialog',
+    size: 'sm',
+    onClosed: jest.fn(),
+    title: 'Hello World!',
+    showCloseButton: true,
+  };
+});
+
+test('Modal renders', () => {
+  const root = render(
+    <Modal {...props}>My Content</Modal>
+  );
+
+  const modal = root.getByRole('dialog');
+  expect(modal.style.getPropertyValue('display')).toBe('block');
+  expect(modal.classList).toContain('my-modal-dialog');
+  expect(modal.classList).toContain('modal');
+
+  const document = root.getByRole('document');
+  expect(document.classList).toContain('modal-sm');
+
+  const header = root.getByRole('heading');
+  expect(header.textContent).toContain('Hello World!');
+
+  const body = root.getByText('My Content');
+  expect(body).toBeTruthy();
+
+  const closeButton = root.getByLabelText('Close');
+  expect(closeButton).toBeTruthy();
+});
+
+test('Closing the Modal', () => {
+  const root = render(
+    <Modal {...props}>My Content</Modal>
+  );
+
+  const closeButton = root.getByLabelText('Close');
+  fireEvent.click(closeButton);
+
+  expect(props.onClosed).toBeCalled();
+});
+
+test('Modal is not shown', () => {
+  const root = render(
+    <Modal {...props} isOpen={false}>My Content</Modal>
+  );
+  expect(root.container.textContent).toBeFalsy();
```

Comment:
```suggestion
  expect(root.container.querySelectorAll('.some-class')).toHaveLength(0);
```
This is how we do this sort of thing elsewhere

Reply:
:-) That guidance also means you should be using things like `await screen.findByRole()` to sort out all the async fun present in javascript.  You're more than welcome to use that if you want, though container.querySelector() has been used throughout our codebase because it's far easier to work with.

The fact that container.querySelector() isn't working is kinda concerning, probably means that your old test wasn't actually testing what you thought it was

Testing negatives is hard. You should be able to use something like this to wait for everything to load:

```
import { render, act } from '@testing-library/react';
// ...
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 100));
  });
```